### ISSUE TITLE
Version without "unsafe" import for appengine?

### ISSUE THREAD

**Issue by jonathaningram** [JalfResi/justext#30]
Is it possible to add a version of the `defaultTemplate.go` and `detailedTemplate.go` files that don't use the `unsafe` package? If it's technically possible, this can be done with Go build tags.

The underlying issue is that one can't import your package in appengine apps:

```
go-app-builder: Failed parsing input: parser: bad import "unsafe" in github.com/JalfResi/justext/classifyParagraphs.go from GOPATH
```

**Comment by JalfResi** [comment]
It's been a very long time since I wrote this so I'm not even sure
offhand why it uses the unsafe package; I can't think of any reason why
it should need it. I'll try to get a look at it this week, but I am more
than happy to accept pull requests!

On Fri, Aug 25, 2017, at 07:37 AM, Jonathan Ingram wrote:
> Is it possible to add a version of the defaultTemplate.go and
> detailedTemplate.go files that don't use the unsafe package? If it's
> technically possible, this can be done with Go build tags.> The underlying issue is that one can't import your package in
> appengine apps:


> go-app-builder: Failed parsing input: parser: bad import "unsafe" in
> github.com/JalfResi/justext/classifyParagraphs.go from GOPATH
>> — You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub[1], or mute the
> thread[2].> 


Links:

  1. https://github.com/JalfResi/justext/issues/30

**Comment by jonathaningram** [comment]
No worries, old code gets a bit that way!

If I can figure out what the current code is doing and provide an alternative for appengine, I'll definitely PR it.

**Comment by jonathaningram** [comment]
Looks like we should be able use go-bindata to generate a version of those templates without the unsafe package per:

https://github.com/jteeuwen/go-bindata#lower-memory-footprint

**Comment by JalfResi** [comment]
Hi @jonathaningram 

Apologies, but I'm not able to get to this anytime soon. If you can produce a PR I'll happily accept it.

**Comment by jonathaningram** [comment]
Hey @JalfResi no worries, I'll close it. I started doing it, but got into a world of mess because I tried using the latest version of the go-bindata tool and I didn't want to revert back to the previous version! Will revisit if it becomes a blocker for us. FYI, in the meantime I just used build tags on our end to effectively bypass this lib and just used a passthru function that does nothing. Cheers, Jon.
